src/tools/tracejob.c

```diff
@@ -481,13 +488,35 @@ parse_log(FILE *fp, char *job, int ind)
 
 			if (tmp.date != NULL) {
 				log_lines[ll_cur_amm].date = strdup(tmp.date);
-				if (sscanf(tmp.date, "%d/%d/%d %d:%d:%d", &tms.tm_mon, &tms.tm_mday, &tms.tm_year, &tms.tm_hour, &tms.tm_min, &tms.tm_sec) != 6)
-					log_lines[ll_cur_amm].date_time = -1;	/* error in date field */
-				else {
-					if (tms.tm_year > 1900)
-						tms.tm_year -= 1900;
-					tms.tm_mon--;         /* The number of months since January, in the range 0 to 11 for mktime */
-					log_lines[ll_cur_amm].date_time = mktime(&tms);
+				if ((ind != IND_ACCT) && (strchr(tmp.date, '.'))) {
+					if (sscanf(tmp.date, "%d/%d/%d %d:%d:%d.%ld", &tms.tm_mon, 
+					    &tms.tm_mday, &tms.tm_year, &tms.tm_hour, &tms.tm_min, 
+					    &tms.tm_sec, &(log_lines[ll_cur_amm].highres_ts)) != 7) {
+						log_lines[ll_cur_amm].date_time = -1;	/* error in date field */
+						log_lines[ll_cur_amm].highres_ts = -1;
+					} else {
+						has_high_res_timestamp = 1;
+						if (tms.tm_year > 1900)
+							tms.tm_year -= 1900;
+						/* The number of months since January, 
+ 						 * in the range 0 to 11 for mktime()
+ 						 */
+						tms.tm_mon--;
+						log_lines[ll_cur_amm].date_time = mktime(&tms);
+					}
+				} else {
+					if (sscanf(tmp.date, "%d/%d/%d %d:%d:%d", &tms.tm_mon, &tms.tm_mday, 
+					    &tms.tm_year, &tms.tm_hour, &tms.tm_min, &tms.tm_sec) != 6) {
+						log_lines[ll_cur_amm].date_time = -1;	/* error in date field */
+						log_lines[ll_cur_amm].highres_ts = -1;
```

Comment:
Does it make sense to make -1  a macro?

Reply:
Probably.  It's set to -1 up front and then the -1 is checked as a value.  If we set it to -1 and then checked >= 0 that'd be one thing.  Since we are specifically looking for -1, we should use a macro.